src/vcModals.cpp

```diff
@@ -619,6 +619,83 @@ void vcModals_DrawProfile(vcState* pProgramState)
+  float buttonWidth = 80.f;
+  ImGui::SetNextWindowSize(ImVec2(width, height), ImGuiCond_Always);
+
+  if (ImGui::BeginPopupModal(profile, nullptr, ImGuiWindowFlags_NoResize))
+  {
+    pProgramState->modalOpen = true;
+
+    if (pProgramState->sessionInfo.isOffline)
+    {
+      ImGui::TextUnformatted(vcString::Get("modalProfileOffline"));
+    }
+    else
+    {
+      ImGuiInputTextFlags passwordFieldFlags = ImGuiInputTextFlags_Password;
+
+      ImGui::InputText(vcString::Get("modalProfileCurrentPassword"), &pProgramState->changePassword.currentPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::Separator();
+      ImGui::InputText(vcString::Get("modalProfileNewPassword"), &pProgramState->changePassword.newPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::InputText(vcString::Get("modalProfileReEnterNewPassword"), &pProgramState->changePassword.newPasswordConfirm[0], vcMaxPathLength, passwordFieldFlags);
+    }
+
+    ImGui::Separator();
+    
+    if (ImGui::Button(vcString::Get("modalProfileConfirmNewPassword"), ImVec2(130.0f, 0)) || vcHotkey::IsPressed(vcB_Cancel))
+    {
+      udJSON changePasswordData;
+      changePasswordData.Set("username = \"%s\"", pProgramState->settings.loginInfo.username);
+      changePasswordData.Set("oldpassword = \"%s\"", pProgramState->changePassword.currentPassword);
+      changePasswordData.Set("password = \"%s\"", pProgramState->changePassword.newPassword);
+      changePasswordData.Set("passwordConfirm = \"%s\"", pProgramState->changePassword.newPasswordConfirm);
+
+      const char *pUpdatePasswordString = nullptr;
+      changePasswordData.Export(&pUpdatePasswordString, udJEO_JSON);
+
+      const char *pResult = nullptr;
+      vdkError result = vdkServerAPI_Query(pProgramState->pVDKContext, "v1/user/updatepassword", pUpdatePasswordString, &pResult);
+      if (result == vE_Success)
+      {
+        udJSON resultData;
+        resultData.Parse(pResult);
+
+        if (resultData.Get("success").AsBool())
+        {
+          ImGui::CloseCurrentPopup();
+          memset(pProgramState->changePassword.currentPassword, 0, sizeof(pProgramState->changePassword.currentPassword));
+          memset(pProgramState->changePassword.newPassword, 0, sizeof(pProgramState->changePassword.newPassword));
+          memset(pProgramState->changePassword.newPasswordConfirm, 0, sizeof(pProgramState->changePassword.newPasswordConfirm));
+          memset(pProgramState->changePassword.message, 0, sizeof(pProgramState->changePassword.message));
+        }
+        else
+        {
+          udStrcpy(pProgramState->changePassword.message, resultData.Get("message").AsString());
+        }
+      }
+    }
+    
+    ImGui::SameLine();
+
+    if (ImGui::Button(vcString::Get("popupClose"), ImVec2(buttonWidth, 0)) || vcHotkey::IsPressed(vcB_Cancel))
+      ImGui::CloseCurrentPopup();
+
+    ImGui::SameLine();
+    ImGui::TextColored(ImVec4(1, 0, 0, 1), pProgramState->changePassword.message);
```

Comment:
This is the line failing the builds on Android and Linux, it needs to be `"%s", pProgramState->changePassword.message`